pulsar-broker/src/test/java/org/apache/pulsar/client/impl/TableViewTest.java

```diff
@@ -129,6 +129,23 @@ private Set<String> publishMessages(String topic, int count, boolean enableBatch
         return keys;
     }
 
+    @Test
+    public void testRefreshAPI() throws Exception {
+        String topic = "persistent://public/default/testRefreshAPI";
+        admin.topics().createPartitionedTopic(topic, 3);
+        int count = 50;
+        Set<String> keys = this.publishMessages(topic, count, false);
+        @Cleanup
+        TableView<byte[]> tv = pulsarClient.newTableViewBuilder(Schema.BYTES)
+                .topic(topic)
+                .create();
+        // Call the refresh API to update the tableview. After this operation,
+        // there is no need to use 'Awaitility.await()' when assert the size and keys of 'tv'.
```

Comment:
You should publish more messages since creating the table view will also load all the data from the server side before it is available for users.

We should also consider stopping or slowing the `readTailMessages` method to make sure the refreshed data is contributed by the refresh API, not the `readTailMessages` method. Otherwise, you don't if this test can protect the newly added refresh API to avoid break changes.